src/SFML/Audio/SoundBuffer.cpp

```diff
@@ -213,7 +207,13 @@ SoundBuffer& SoundBuffer::operator=(const SoundBuffer& right)
     std::swap(m_samples, temp.m_samples);
     std::swap(m_buffer, temp.m_buffer);
     std::swap(m_duration, temp.m_duration);
-    std::swap(m_sounds, temp.m_sounds); // swap sounds too, so that they are detached when temp is destroyed
+
+    // Reattach sounds that use this buffer so they get bound to the new OpenAL buffer
+    for (Sound* soundPtr : m_sounds)
+    {
+        soundPtr->stop();
+        soundPtr->reattachBuffer();
+    }
```

Comment:
What's the point of reattaching if we previously assert when the m_sounds isn't empty?

Reply:
At this point in the code, sounds are still correctly related to the SoundBuffer, but underlying OpenAL sources are still incorrectly bound to the previous OpenAL buffer. Calling reattachBuffer here makes the sounds bind to the new OpenAL buffer whose id was put in the SoundBuffer's m_buffer member.

Reply:
Sorry, I now realize I didn't quite answer the question. So, we assert when m_sounds is empty in SoundBuffer destructor. Here `temp` is about to get desctucted but it has not sounds attached. I don't see any problem about iterating on m_sounds here. m_sounds content is not touched by the assignment operator.

Reply:
I'm sorry, I somehow didn't have all the code expanded on GitHub and made some wrong connections (read the beginning of `~SoundBuffer()` and thought this code was at the end of the function...) 🤦